lib/ripper-tags/json_formatter.rb

```diff
@@ -16,19 +16,37 @@ def stream_format?
 
     def with_output
       super do |true_out|
-        buffer = []
-        yield buffer
+        @buffer ||= []
+        yield @buffer
+        @buffer.uniq!
 
         if stream_format?
-          buffer.each { |tag| true_out.puts ::JSON.dump(tag) }
+          @buffer.each { |tag| true_out.puts ::JSON.dump(tag) }
         else
-          true_out.write ::JSON.dump(buffer)
+          true_out.write ::JSON.dump(@buffer)
         end
       end
     end
 
+    def prepare_output(filename, &block)
+      if options.append && File.readable?(filename)
+        @buffer = JSONTagsProcessor.new(filename, stream_format?).read
+      end
+      super
+    end
+
     def write(tag, buffer)
       buffer << tag
     end
   end
+
+  JSONTagsProcessor = Struct.new(:filename, :stream) do
+    def read
+      if stream
+        File.readlines(filename).map { |line| ::JSON.parse(line, :symbolize_names => true) }
```

Comment:
In "stream" mode, appending to an existing tag file is not very useful, since stream mode is mostly designed to be outputting to stdout or to a pipe instead to a file on disk.

In fact, I would vote that `--apend` is not supported in JSON mode at all.